### DISCUSSION TITLE
Take into account the derivative of my control variable in the model

### DISCUSSION THREAD

**Discussion by polytech1** [dynamicslab/pysindy#522] [Q&A]
Hey,

I am using PySindy to model multivariate time series where there is a control variable, that is, it's a variable which fully depends on human action and does not need to be modeled. This variable usually transitions quickly from a low level to a higher one and vice-versa, or sometimes evolves by increments. So, I used PySINDY-c with some `u` variable in the model. 

If I want to involve its derivative(s) in the model, I have no other choice than "manually" differientating the command and pass it as a second control variable, right ? Or is there better way to proceed ? 

Thanks

**Comment by Jacob-Stevens-Haas** [comment]
That is correct.

**Reply by polytech1** [reply]
@Jacob-Stevens-Haas So no direct way to know "easily" through PySINDY up to which order I should consider the derivatives of the control variable to accurately represent my data ? Would you have some workaround for this ?

**Reply by Jacob-Stevens-Haas** [reply]
No workaround, just estimate the derivatives using either `pysindy` or `derivative`'s smoothing/differentiating.  Usually I would say 0 derivatives unless you have reason to believe they matter.  E.g. your control is the position of a magnet and it's derivative causes an electric field.
